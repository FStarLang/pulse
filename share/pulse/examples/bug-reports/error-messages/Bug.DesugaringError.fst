module Bug.DesugaringError

#lang-pulse
open Pulse.Lib.Pervasives

[@@expect_failure]
fn test (x:nat) (y:nat) 
requires emp
ensures emp
{
    a ; an ()
}

[@@expect_failure]
ghost
fn test2 (x:something)
requires emp
ensures emp
{
    ()
}

let id (#a:Type) (x y:a) = x

[@@expect_failure]
fn test3 (x:nat) (y:bool)
requires emp
returns z:nat
ensures emp
{
    id x y
}

fn f (x: nat)
requires emp
ensures emp
{
  ()
}

[@@expect_failure]
fn test4 ()
requires emp
ensures emp
{
  f true
}
