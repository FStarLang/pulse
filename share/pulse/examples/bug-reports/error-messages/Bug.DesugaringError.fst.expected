>> Got issues: [
* Error 168 at Bug.DesugaringError.fst(7,0-7,0):
  - Failed to desugar pulse declaration

* Error 72 at Bug.DesugaringError.fst(11,4-11,5):
  - Failed to desugar Pulse term
  - Identifier not found: [a]

>>]
>> Got issues: [
* Error 168 at Bug.DesugaringError.fst(15,0-15,0):
  - Failed to desugar pulse declaration

* Error 72 at Bug.DesugaringError.fst(16,12-16,21):
  - Failed to desugar Pulse term
  - Identifier not found: [something]

>>]
>> Got issues: [
* Error 228 at Bug.DesugaringError.fst(31,4-31,4):
  - Could not check term: Bug.DesugaringError.id x y

* Error 189 at Bug.DesugaringError.fst(31,9-31,10):
  - Expected expression of type Prims.nat got expression y of type Prims.bool
  - Raised within Tactics.refl_instantiate_implicits
  - See also Bug.DesugaringError.fst(31,4-31,10)

>>]
>> Got issues: [
* Error 228 at Bug.DesugaringError.fst(46,2-46,2):
  - Could not check term: Bug.DesugaringError.f true

* Error 189 at Bug.DesugaringError.fst(46,4-46,8):
  - Expected expression of type Prims.nat got expression true of type Prims.bool
  - Raised within Tactics.refl_instantiate_implicits

>>]
Verified module: Bug.DesugaringError
All verification conditions discharged successfully
